### PR TITLE
[TASK-8] As a user, I can view the lazy loading screen

### PR DESCRIPTION
**Pull request by rafayet-monon**
Task Card - https://github.com/rafayet-monon/surveyor/projects/1#card-43745447

## What happened
This loading screen will be used for a smoother transition of the home page loading.
 
## Insight
- [x] Create Lazy Loading Component
- [x] Add Lazy Loading Styles 

## Proof Of Work
- **Animation design in [Figma](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?node-id=6%3A168)** 

- **Implemented Design**

React App
 
- **Responsiveness**

Screenshot 2020-09-15 at 7 03 54 PM

### CODE REVIEW

File: src/stylesheets/screens/lazy-loader.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    border-radius: rem(12px);
+    height: rem(18px);
+    width: rem(207px);
+  }
+
+  &__navigation {
+    margin-left: auto;
+    order: 2;
+    position: relative;
+    @include media-breakpoint(map-get($grid, 'desktop')) {
```
Comment (namtx) [inline src/stylesheets/screens/lazy-loader.scss]: the value for `desktop` and `mobile` are the same?

File: src/stylesheets/screens/lazy-loader.scss [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    margin-top: 14px;
+  }
+
+  &__day .react-loading-skeleton {
+    border-radius: 12px;
+    height: rem(18px);
+    width: rem(90px);
+  }
+
+  &__survey-img {
```
Comment (namtx) [inline src/stylesheets/screens/lazy-loader.scss:57]: we should avoid using abbr in class name IMO

File: src/stylesheets/screens/lazy-loader.scss [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  }
+
+  &__date .react-loading-skeleton {
+    border-radius: rem(12px);
+    height: rem(18px);
+    width: rem(117px);
+  }
+
+  &__day {
+    margin-top: 14px;
```
Comment (namtx) [inline src/stylesheets/screens/lazy-loader.scss]: everything should be in `rem` I think

File: src/containers/LazyLoader/loader.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';
+
+const Loader = () => {
+  return (
+    <SkeletonTheme color="#313236" highlightColor="#49494D">
+      <div className="skeleton-header">
+        <span className="skeleton__menu-bar">
```
Comment (namtx) [inline src/containers/LazyLoader/loader.js]: as we are following the BEM, should it be named as `skeleton-header__menu-bar`?

File: src/stylesheets/screens/lazy-loader.scss (hunk 1 above)
Comment (rafayet-monon) [reply src/stylesheets/screens/lazy-loader.scss]: For `desktop` and `tablet`, I put the same value of `bottom`. For tried the mobile-first approach as said in the compass.

File: src/containers/LazyLoader/loader.js (hunk 4 above)
Comment (rafayet-monon) [reply src/containers/LazyLoader/loader.js]: Fixed in [6c44c1b](https://github.com/rafayet-monon/surveyor/pull/9/commits/6c44c1bd55bc12370c81af6d4a2cf087d87dc30f)

File: src/containers/LazyLoader/loader.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';
+
+const Loader = () => {
+  return (
+    <SkeletonTheme color="#313236" highlightColor="#49494D">
+      <div className="skeleton-header">
```
Comment (Lahphim) [inline src/containers/LazyLoader/loader.js:11]: @rafayet-monon I think for this naming `skeleton` of class, you're trying to map with the component name, right?
But can we make it in a different name which more relates to its component?

File: src/stylesheets/mixins/media-breakpoint.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+@mixin media-breakpoint($size) {
```
Comment (Lahphim) [inline src/stylesheets/mixins/media-breakpoint.scss]: @rafayet-monon Why don't we use mixin from bootstrap?

File: src/containers/LazyLoader/loader.js (hunk 5 above)
Comment (rafayet-monon) [reply src/containers/LazyLoader/loader.js:11]: Can you please give an example @Lahphim ? 🤔

File: src/stylesheets/screens/lazy-loader.scss (hunk 1 above)
Comment (rafayet-monon) [reply src/stylesheets/screens/lazy-loader.scss]: @namtx Should I keep it as it is?

File: src/stylesheets/mixins/media-breakpoint.scss (hunk 6 above)
Comment (rafayet-monon) [reply src/stylesheets/mixins/media-breakpoint.scss]: Fixed in [89e392b](https://github.com/rafayet-monon/surveyor/pull/9/commits/89e392b80eede7fd5a80ecacb83653f4d7d838be)

File: src/containers/LazyLoader/loader.js (hunk 5 above)
Comment (rafayet-monon) [reply src/containers/LazyLoader/loader.js:11]: @Lahphim should I keep it as it is?

**Review by Lahphim — APPROVED** [review]

File: src/stylesheets/screens/lazy-loader.scss (hunk 1 above)
Comment (namtx) [reply src/stylesheets/screens/lazy-loader.scss]: If we don't have different values for each screen, we should omit the `media-breakpoint` I think.

Comment (rafayet-monon) [reply src/stylesheets/screens/lazy-loader.scss]: The thing is, for mobile screens I'm using no `bottom` attribute. I'm using it for desktop and tablet which has the same `bottom` value. What should be done in these cases? @namtx

Comment (namtx) [reply src/stylesheets/screens/lazy-loader.scss]: so just `@include media-breakpoint(map-get($grid, 'tablet')) {}` is enough

File: src/stylesheets/screens/lazy-loader.scss [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    width: rem(36px);
+  }
+}
+
+.skeleton-details {
+  display: flex;
+  flex-direction: column;
+  max-width: rem(704px);
+  padding: rem(50px 32px);
+  @include media-breakpoint-up(md) {
```
Comment (rafayet-monon) [inline src/stylesheets/screens/lazy-loader.scss:37]: @namtx breakpoint issue changed here as well

File: src/stylesheets/screens/lazy-loader.scss [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    border-radius: rem(12px);
+    height: rem(18px);
+    width: rem(207px);
+  }
+
+  &__navigation {
+    margin-left: auto;
+    order: 2;
+    position: relative;
+    @include media-breakpoint-up(md) {
```
Comment (rafayet-monon) [inline src/stylesheets/screens/lazy-loader.scss:91]: @namtx changed in [be41e1b](https://github.com/rafayet-monon/surveyor/pull/9/commits/be41e1bba2eda039eb95010cd29853aec36a379a)

Comment (namtx) [reply src/stylesheets/screens/lazy-loader.scss:91]: 👍

File: src/containers/LazyLoader/loader.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';
+
+const Loader = () => {
+  return (
+    <SkeletonTheme color="#313236" highlightColor="#49494D">
```
Comment (namtx) [inline src/containers/LazyLoader/loader.js]: How about split these color values into `constants`?

Comment (rafayet-monon) [reply src/containers/LazyLoader/loader.js]: Changed in [10ef0d6](https://github.com/rafayet-monon/surveyor/pull/9/commits/10ef0d66435899145581b37dbe46f8721d9749fc)

**Review by namtx — APPROVED** [review]

**Pull request merged**